redlines/utils/pdf_processor.py

```
@@ -0,0 +1,161 @@
+    # ------------------------------------------------------------------
+    async def _html_document_to_pdf(self, html_document: str, prefer_backend: Optional[str]) -> bytes:
+        async with self._semaphore:
+            backends = self._ordered_backends(prefer_backend)
+            last_exception: Optional[Exception] = None
+            for backend in backends:
+                try:
+                    if backend == 'weasyprint' and self.weasyprint_enabled:
+                        return await self._render_weasyprint(html_document)
+                    if backend == 'xhtml2pdf' and self.xhtml2pdf_enabled:
+                        return await self._render_xhtml2pdf(html_document)
+                except Exception as exc:  # pragma: no cover - backend specific failures
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)